extension/persistence/relational-jdbc/src/main/java/org/apache/polaris/extension/persistence/relational/jdbc/RelationalJdbcConfiguration.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.extension.persistence.relational.jdbc;
+
+import io.smallrye.config.ConfigMapping;
+import java.util.Optional;
+
+@ConfigMapping(prefix = "polaris.persistence.relational.jdbc")
```

Comment:
I believe this should be kept-inside quarkus-specific module... That is, the JDBC module can declare a config interface, but quarkus module will have to extend it and provide config annotations.

Reply:
I see, i was in the same delima as i had similar comments in my pr.
1. Presently there is no quarkus/common to share this between quarkus/admin and quarkus/service so will need to create one if we wanna move this to there 
2. Eclipse link configs are also defined this way. 

If IIUC the concern here is mostly from best practice perspective as we don't wanna config name here ? and each framework can define there own Conf ? do we want to define that exp ? we would want to be consistent. 

Let me know if you strongly feel about it, happy to refactor !

Reply:
2: I do not think prior art in EclipseLink is a valid argument here, because EclipseLink is deprecated :)

Reply:
1: Why not a new shared module for the pieces that help plugging JDBC persistence into Quarkus?

I believe having that module would be preferable to pulling in server runtime specific dependencies (`@ConfigMapping`) into the JDBC persistence module. JDBC declares what config it needs via this interface. Runtime modules provide this config using whatever methods are appropriate in each case. If we want to share code between admin and server - we add a shared module.

Reply:
Having the `@ConfigMapping` annotation here via the smallrye-config dependency is fine IMO

Reply:
`@ConfigMapping` conveys certain expectation as to how the config values can be provided, but the JDBC module alone does not have a runtime component that provides those values.

If it were to be reused in a downstream project, the present of this annotation is likely to interfere with the project-specific approach to managing configuration.

Reply:
> @ConfigMapping conveys certain expectation as to how the config values can be provided, but the JDBC module alone does not have a runtime component that provides those values.

Thats certainly one way to look at it other is to look at it in the perspective of every runtime should have consistent config name irrespective if it's spring boot or quarkus having a conf in this module we define it for consistency.

Nevertheless i believe you feel more strongly about this pattern and I totally respect that, than i think of the other pattern. Hence i made the changes to make them in separate module ! 
Again i am very new to quarkus ! may be this is a well established pattern in these frameworks, hence I might not be able to get it, thank you so much for patiently explaining !